lib/matplotlib/_types.py

```diff
@@ -0,0 +1,179 @@
+    ``linewidth/2``, away from any point of the center line. However, this
+    results in corners appearing "rounded", which may not be the desired
+    behavior if you are drawing, for example, a polygon or pointed star.
+
+    Matplotlib provides three options for drawing the corners between adjacent
+    segments. In short:
+
+    - *miter* is the "arrow-tip" style. Each boundary of the filled-in area
+      will extend in a straight line parallel to the tangent vector of the
+      centerline at the point it meets the corner, until they meet in a
+      sharp point.
+    - *round* stokes every point within a radius of ``linewidth/2`` of the
+      center lines.
+    - *bevel* is the "squared-off" style. It can be thought of as a rounded
+      corner where the "circular" part of the corner has been cut off.
+
+    .. note::
+
+        The *miter* option can be controlled further by specifying a "miter
+        limit", which specifies how long a miter tip can get before it is
+        automatically "bevel"ed off. Matplotlib does not currently expose a
+        ``miterlimit`` parameter to the user, and most backends simply use the
+        upstream default value. For example, the PDF backend assumes the
+        default value of 10 specified by the PDF standard, while the SVG
+        backend does not even specify the miter limit, resulting in a default
+        value of 4 per the SVG specification.
+
+        A more detailed description of the effect of a miter limit can be found
+        in the `Mozilla Developer Docs
+        <https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit>`_
+    """
+
+    miter = auto()
+    round = auto()
+    bevel = auto()
+
+    def __init__(self, s):
+        s = _deprecate_case_insensitive_join_cap(s)
+        Enum.__init__(self)
+
+    @staticmethod
+    def demo():
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        def plot_angle(ax, x, y, angle, style):
+            phi = np.radians(angle)
+            xx = [x + .5, x, x + .5*np.cos(phi)]
+            yy = [y, y, y + .5*np.sin(phi)]
+            ax.plot(xx, yy, lw=12, color='tab:blue', solid_joinstyle=style)
+            ax.plot(xx, yy, lw=1, color='black')
+            ax.plot(xx[1], yy[1], 'o', color='tab:red', markersize=3)
+
+        fig, ax = plt.subplots(figsize=(8, 6))
+        ax.set_title('Join style')
+        for x, style in enumerate(['miter', 'round', 'bevel']):
+            ax.text(x, 5, style)
+            for y, angle in enumerate([20, 45, 60, 90, 120]):
+                plot_angle(ax, x, y, angle, style)
+                if x == 0:
+                    ax.text(-1.3, y, f'{angle} degrees')
+        ax.set_xlim(-1.5, 2.75)
+        ax.set_ylim(-.5, 5.5)
+        ax.set_axis_off()
+        fig.show()
+JoinStyle.input_description = "{" \
+        + ", ".join([js.name for js in JoinStyle]) \
+        + "}"
+
+
+class CapStyle(str, _AutoStringNameEnum):
```

Comment:
I'm not particularly happy with the overall appearance of the docs (https://48026-1385122-gh.circle-artifacts.com/0/doc/build/html/api/_types.html#matplotlib._types.CapStyle). It looks a bit cluttered:

- The supported values blend into the bulk of text and are hard to find.
- The demo image should be below the list of values.
- There's a lot of redundancy: Available values are mentioned 4 times: list including descriptions, in the example plot, in `input_description`, as attributes of the Enum.
- While I see the use-case `demo()` still feels a bit out of place. You can keep it for now, but please give it a docstring.

Reply:
I agree that the output looks cluttered. I've:

- edited the docstrings to emphasize supported values, by making it a section
- moved the image to the bottom
- removed all automatic member documentation from the enum to prevent any redundacy

Reply:
The above are good steps in the right direction, but I'm still somewhat worried that the values are difficult to find. They blend into the paragraphs of the bullet list.

Visibility could be further enhanced with a definition list. E.g.

~~~
    **Supported values:**

    .. rst-class:: value-list

       'miter'
          the "arrow-tip" style. Each boundary of the filled-in area
          will extend in a straight line parallel to the tangent vector of the
          centerline at the point it meets the corner, until they meet in a
          sharp point.
       'round'
          stokes every point within a radius of ``linewidth/2`` of the
          center lines.
       'bevel'
          the "squared-off" style. It can be thought of as a rounded
          corner where the "circular" part of the corner has been cut off.
~~~

Will result in `<dl class="value-list">...</dl>` so that we can CSS-style based on `.value-list`.

This is the lightweight version of making your own `.. value-list::` role, which may or may not be reasonable later on.